src/main/java/net/nextencia/rrdiagram/Server.java

```diff
@@ -0,0 +1,264 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.net.URLDecoder;
+import java.net.URI;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+
+class Helper {
+    public static boolean writeHTTPResponse(HttpExchange httpExchange, int responseCode, String content, String contentType) {
+        try {
+            OutputStream outputStream = httpExchange.getResponseBody();
+            httpExchange.getResponseHeaders().set("Content-Type", contentType);
+            httpExchange.sendResponseHeaders(responseCode, content.length());
+            outputStream.write(content.getBytes());
+            outputStream.flush();
+            outputStream.close();
+            return true;
+        } catch(java.io.IOException e) {
+            e.printStackTrace(System.out);
+            return false;
+        }
+    }
+
+    public static boolean writeHTTPResponse(HttpExchange httpExchange, int responseCode, String content) {
+        return writeHTTPResponse(httpExchange, responseCode, content, "text/plain; charset=us-ascii");
+    }
+}
+
+class BNFHandler implements HttpHandler {
+
+    Logger logger = Logger.getLogger(BNFHandler.class.getName());
+
+    public BNFHandler() {
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+        if("GET".equals(httpExchange.getRequestMethod())) { 
+            handleRequest(httpExchange);
+        }
+    }
+
+    public Map<String, String> parseParameters(HttpExchange exchange) {
+        URI requestedUri = exchange.getRequestURI();
+        String query = requestedUri.getRawQuery();
+        Map<String, String> parameters = new HashMap<String, String>();
+         if (query != null) {
+             String pairs[] = query.split("[&]");
+
+             for (String pair : pairs) {
+                String param[] = pair.split("[=]");
+
+                String key = null;
+                String value = null;
+                if (param.length > 0) {
+                    try {
+                        key = URLDecoder.decode(param[0],System.getProperty("file.encoding"));
+                    } catch (java.io.UnsupportedEncodingException ue) {
+                        ue.printStackTrace();
+                    }
+                }
+
+                if (param.length > 1) {
+                    try {
+                        value = URLDecoder.decode(param[1],System.getProperty("file.encoding"));
+                    } catch (java.io.UnsupportedEncodingException ue) {
+                        ue.printStackTrace();
+                    }
+                }
+                
+                if (key != null && value != null) {
+                    parameters.put(key, value);
+                }
+            }
+        }
+        return parameters;
+    }
+
+    public String cleanString(String input) {
```

Comment:
seems unnecessarily public.  or also, should it be static?  does it belong in this class?

Reply:
We would need it to be public when we add test cases. Will later move to utils class.